tests/sentry/models/test_release.py

```diff
@@ -566,6 +567,48 @@ def test_resolution_support_with_integration(self, mock_sync_status_outbound):
 
         assert Group.objects.get(id=group.id).status == GroupStatus.RESOLVED
 
+    def test_release_commits_updated(self):
+        org = self.create_organization()
+
+        repo = Repository.objects.create(
+            organization_id=org.id,
+            name='test/repo',
+        )
+        commit = Commit.objects.create(
+            organization_id=org.id,
+            repository_id=repo.id,
+            message='Something',
+            key='alksdflskdfjsldkfajsflkslk',
+        )
+        release = Release.objects.create(version='abcdabc', organization=org)
```

Comment:
There is a `create_release()` fixture method that will generate a release, repository and commit if given a `user` parameter. You could then call set_commits() with the new commit and be able to test both the add + remove sides.